### ISSUE TITLE
Use of custom NavigationPage class in Xamarin.Forms crashes in iOS build with System.MissingMethodException: Method 'UIKit.UINavigationBar.set_LargeTitleTextAttributes' not found. 

### ISSUE THREAD

**Issue by productXF** [xamarin/Xamarin.Forms#1379]
### Description
Migrated from https://bugzilla.xamarin.com/show_bug.cgi?id=61019
Use of custom NavigationPage class in Xamarin.Forms crashes in iOS build with following error only in latest XF version :-

Unhandled Exception:
System.MissingMethodException: Method '**UIKit.UINavigationBar.set_LargeTitleTextAttributes**' not found.

### Steps to Reproduce

1. Create sample Xamarin.Forms app with XF v2.5.0.91635 OR with XF v2.5.0.121934
2. Create custom NavigationPage class by extending NavigationPage.
3. Use this custom class to add ContentPages of app in Navigation. On launch app crashes only in iOS with System.MissingMethodException: Method 'UIKit.UINavigationBar.set_LargeTitleTextAttributes'
error.

iOS app is working fine with XF v2.3.3.193.

### Expected Behavior
iOS app should run normal.

### Actual Behavior
iOS app crashes.

### Basic Information

- Version with issue: v2.5.0.121934 and v2.5.0.91635 
- Last known good version: XF v2.3.3.193.
- IDE: Visual Studio 2015 update 3
- Platform Target Frameworks: 
  - iOS:  iOS SDK 10.3
  - Android: 7.1
  - UWP:  NA
- Android Support Library Version: 25.4.0.2
- Nuget Packages:
1. Xamarin.Forms.2.5.0.121934
2. Xamarin Support libs 25.4.0.2

- Affected Devices:

### Environment details as below :-

Microsoft Visual Studio Professional 2015
Version 14.0.25420.01 Update 3
Microsoft .NET Framework
Version 4.6.01055

Installed Version: Professional

LightSwitch for Visual Studio 2015   00322-50050-34434-AA417
Microsoft LightSwitch for Visual Studio 2015

Visual Basic 2015   00322-50050-34434-AA417
Microsoft Visual Basic 2015

Visual C# 2015   00322-50050-34434-AA417
Microsoft Visual C# 2015

Visual C++ 2015   00322-50050-34434-AA417
Microsoft Visual C++ 2015

Visual F# 2015   00322-50050-34434-AA417
Microsoft Visual F# 2015

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub.VisualStudio   2.2.0.11
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Merq   1.1.17-rc (cba4571)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Mono Debugging for Visual Studio   Mono.Debugging.VisualStudio
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Office Developer Tools for Visual Studio 2015 ENU   14.0.23025
Microsoft Office Developer Tools for Visual Studio 2015 ENU

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

SQL Server Data Tools   14.0.60519.0
Microsoft SQL Server Data Tools

TypeScript   1.8.34.0
TypeScript tools for Visual Studio

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.

Xamarin   4.5.0.486 (fec6f88)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin.Android   7.3.1.2 (9dbc4c5)
Visual Studio extension to enable development for Xamarin.Android.

Xamarin.iOS   10.10.0.37 (ad35de4)
Visual Studio extension to enable development for Xamarin.iOS.

**Comment by rmarinho** [comment]
@jassmith i think he updated Xamarin.iOS and/or XCode and now there's a new api.. so this is because he has a CustomNavigationPage, but i think this is a tools issue.

**Comment by jassmith** [comment]
I agree that is likely a tools issue but I dont see how one gets in this state.

**Comment by productXF** [comment]
@rmarinho :- I have not updated anything else apart from Xamarin.Forms version. It was updated from XF v2.3.3.193 to v2.5.0.121934 and v2.5.0.91635. 

@jassmith :- I am using iOS simulator running iOS 10.3(14E269) and Xcode v8.3.2(8E2002).

You can find full environment details in my initial issue **Environment details as below**.

**Comment by jassmith** [comment]
Can you please leave the full stack trace?

**Comment by productXF** [comment]
@jassmith :- I am launching application from Visual Studio 2015 to Mac mini through Mac agent on iOS simulator running iOS 10.3. **Device Log**  does not list iOS simulator, don't know why?  In **Output** console I get only following for crash.

**Unhandled Exception:
System.MissingMethodException: Method 'UIKit.UINavigationBar.set_LargeTitleTextAttributes' not found.**

Did you try reproducing at your end? I have mentioned **Steps to Reproduce** , create sample application with specified XF version & have a custom NavigationPage used in your app. It won't take much of your time.

**Comment by rmarinho** [comment]
iOS10 with xcode 8.3? Can you post all the info you see in the about section for Visual Studio and your Visual Studio for Mac ?

**Comment by rmarinho** [comment]
@productXF can you update xcode to 9.2 and xamarin stable on your mac  ?

**Comment by productXF** [comment]
@rmarinho :- Yes!!! iOS 10.3 with Xcode 8.3.2(8E2002). As I have mentioned earlier I am using Visual Studio 2015 for Windows and connecting to Mac Mini (macOS Sierra v10.12.6) from windows. About section of Windows Visual Studio 2015 has been put up initially in **_Environment details as below_**  section of bug that I have raised.

MonoTouch version is 10.10.0.36 in Mac Mini as per **_/Developer/MonoTouch/Version_** file.
I am not using Visual Studio for Mac, it's a standalone package for Xamarin.iOS.

Requesting you to scroll through all comments from start to end to avoid confusion.

I am using **_shared_** mac mini so I can't update the Xcode , as it may affect other teams.

@davidortinau :-
1] Also we updated Xamarin in PC from 4.5 to 4.7. We saw Visual Studio 2015 was getting hanged specially on cleaning the project. 
I am not sure but you have stopped providing separate installer package for Xamarin.iOS now.
So even if I update Xcode & Xamarin.iOS on mac, I have to update Xamarin.iOS in PC that means full Xamarin but we faced IDE hanging issue in 4.7.

2] Already we have wasted lot of our development time when we upgraded our codebase from XF v2.3.3.193 to XF v2.5.0.9163. Dependent FAB(Floating Action Button) libraries had issues in Android as we have to upgrade the support libs from 23.3.0 to 25.4.0.2. We fixed that in FAB code base but then this iOS issue came.

3] Earlier we had issue with debugging app itself as only our project's breakpoints were not hitting as expected. We created sample apps all apps were hitting the breakpoints properly. We invested time in it to figure out why .mdb files were not attached to the DLL in spite of it were generated in **_bin_** folder. Finally we created fresh project & copied code one by one to get rid of that issue.

To be honest it is really frustrating to invest time in toolkit/ IDE tools issues as suddenly things stopped working even we didn't update anything like in case of debug issue.

We really understand new features & supports will keep coming in latest version but we expect it to be more stable at least in toolkit/IDE tools.

Thanks
